app/code/Magento/Newsletter/Controller/Manage/Save.php

```diff
@@ -74,13 +76,28 @@ public function execute()
                 $customer = $this->customerRepository->getById($customerId);
                 $storeId = $this->storeManager->getStore()->getId();
                 $customer->setStoreId($storeId);
-                $this->customerRepository->save($customer);
-                if ((boolean)$this->getRequest()->getParam('is_subscribed', false)) {
-                    $this->subscriberFactory->create()->subscribeCustomerById($customerId);
-                    $this->messageManager->addSuccess(__('We saved the subscription.'));
+                $isSubscribedState = $customer->getExtensionAttributes()
```

Comment:
This whole section is still a bit hard to read/understand do you think that there would be a way to make it a bit easier to read? Maybe moving some of the logic to new private methods and splitting out some of the if/else so it is not so nested. What do you think?

Reply:
Hi @dmanners 
Thats a reasonable proposition, cause I agree that this part of newsletter module is not the best, but for now I will leave it as it is. If I will find enough time I will come back to it later.